### ISSUE TITLE
New log server/client feature does not compile on HP-UX with HP aCC

### ISSUE THREAD

**Issue by michael-o** [sudo-project/sudo#60]
Trying from master on HP-UX 11.31:
```
export PREFIX=/opt/ports
export LIBDIR=$PREFIX/lib/hpux32
export SYSCONFDIR=$PREFIX/etc
export CC=/opt/aCC/bin/aCC
export CXX=/opt/aCC/bin/aCC
export CONFIGURE="./configure --prefix=$PREFIX --libdir=$LIBDIR"
export CPPFLAGS="-I$PREFIX/include"
export LDFLAGS="-L$LIBDIR"
$CONFIGURE --sysconfdir=$SYSCONFDIR --disable-root-sudo --disable-sasl --disable-pam-session \
  --with-env-editor --without-ldap --disable-shared-libutil --enable-static-sudoers \
  --enable-gss-krb5-ccache-name
```
gives me errors like this:
```
"./iolog_client.c", line 736: error #2136: struct "_ClientMessage" has no
          field "hello_msg"
      client_msg.hello_msg = &hello_msg;
                 ^
```

The reason is that anonymous unions are not standard (ANSI) C. I have reported the same issue with GNU coreutils, see:
* https://stackoverflow.com/a/3228172/696632
* https://lists.gnu.org/archive/html/bug-coreutils/2019-05/msg00019.html

Here is a working patch:
[anon-union.patch.txt](https://github.com/sudo-project/sudo/files/5175849/anon-union.patch.txt)

**Comment by millert** [comment]
This is code generated by protoc-c, which makes patching things more difficult.  I should be able to post-process things to automatically add the union name.  Anonymous unions are a common extension and were standardized in the C11 standard, which is now 9 years old but vendor compilers tend to be out of date.

**Comment by millert** [comment]
Fixed by eaa95acb316696283a740c4aa67ab92ac405c831

**Comment by michael-o** [comment]
Thank you very much. HP aCC does not support this extension. C11 is horrible, have been fighting with this for CMake too. I prefer rather to stick to ANSI C90/C99 and not rely on compiler specifics.
